contracts/pool_stable/src/contract.rs

```diff
@@ -349,6 +351,7 @@ impl StableLiquidityPoolTrait for StableLiquidityPool {
             )
             .to_u128()
             .expect("Pool stable: provide_liquidity: conversion to u128 failed");
+
```

Comment:
```suggestion
    // Calculate change in invariant
    let delta_d = new_invariant.to_u128().expect("Pool stable: provide_liquidity: conversion to u128 failed") - initial_invariant;

    // Calculate minimum acceptable change considering slippage
    let min_delta_d = delta_d * (10000 - custom_slippage as u128) / 10000;

    // Ensure delta_d is within slippage tolerance
    if delta_d < min_delta_d {
        log!(env, "Pool Stable: ProvideLiquidity: Slippage exceeded");
        panic_with_error!(env, ContractError::SlippageExceeded);
    }
```

I'm still not entirely sure if that formula is not an oversimplification.

Can you contract the coinspect team and inquire about that?